pkg/operator/hive/hiveadmission.go

```diff
@@ -311,6 +323,103 @@ func (r *ReconcileHiveConfig) deployFeatureGatesConfigMap(hLog log.FieldLogger,
 	return computeConfigHash(cm), nil
 }
 
+// allowedContracts is the list of operator whitelisted contracts that hive will accept
+// from CRDs.
+var allowedContracts = sets.NewString(
+	hivecontractsv1alpha1.ClusterInstallContractLabelKey,
+)
+
+func (r *ReconcileHiveConfig) deploySupportedContractsConfigMap(hLog log.FieldLogger, h resource.Helper, instance *hivev1.HiveConfig) (string, error) {
+	cm := &corev1.ConfigMap{}
+	cm.Name = supportedContractsConfigMapName
+	cm.Namespace = getHiveNamespace(instance)
+	cm.Data = make(map[string]string)
+
+	crdList := &apiextv1beta1.CustomResourceDefinitionList{}
+	if err := r.Client.List(context.TODO(), crdList); err != nil {
+		hLog.WithError(err).Error("error getting crds for collect contract implementations")
+		return "", err
+	}
+	supported := map[string][]contracts.ContractImplementation{}
+	for _, crd := range crdList.Items {
+		// collect all the possible implementations from this crd
+		var impls []contracts.ContractImplementation
+		for _, version := range crd.Spec.Versions {
+			impls = append(impls, contracts.ContractImplementation{
+				Group:   crd.Spec.Group,
+				Version: version.Name,
+				Kind:    crd.Spec.Names.Kind,
+			})
+		}
+
+		for label, val := range crd.ObjectMeta.Labels {
+			if strings.HasPrefix(label, "contracts.hive.openshift.io") &&
```

Comment:
Going to be  a bit of a pain to get this label onto CRDs when you generate them with kubebuilder. I think it's going to require kustomize as a post step?